### PR TITLE
api gateway and service registry added

### PR DESCRIPTION
**Pull request by faad555**
The service registry will keep the record of all the services e.g auth service, post service, profile service etc
Api-gateway communicate with services via service registry.
So any request comes in to the Api-gateway, the API-gateway checks with Service registry if the service is available and On. Then it will hit the related microservice. so if there's no service registered in the "Service registry" the API-gateway wont be able to communicate with microservice.

### CODE REVIEW

File: Microservices/api-gateway/pom.xml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<parent>
+		<groupId>org.springframework.boot</groupId>
+		<artifactId>spring-boot-starter-parent</artifactId>
+		<version>3.1.2</version>
+		<relativePath/> <!-- lookup parent from repository -->
+	</parent>
+	<groupId>com.fawad</groupId>
```
Comment (M-Byte480) [inline Microservices/api-gateway/pom.xml:11]: Change this to cs4337.group-8

File: Microservices/api-gateway/pom.xml [hunk 2]
```diff
@@ -0,0 +1,61 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<parent>
+		<groupId>org.springframework.boot</groupId>
```
Comment (M-Byte480) [inline Microservices/api-gateway/pom.xml:6]: The parent should inherit from our parent:
    <parent>
        <groupId>cs4337.group-8</groupId>
        <artifactId>parent-microservices</artifactId>
        <version>0.0.1-SNAPSHOT</version>
        <relativePath>../pom.xml</relativePath>
    </parent>

**Review by M-Byte480 — CHANGES_REQUESTED** [review]

**Comment by Silver-Golden** [comment]
How does this relate to #37 ?  
Should both PR's be open?

**Review by M-Byte480 — APPROVED** [review]

**Review by Silver-Golden — APPROVED** [review]
Assuming #37 is no longer required this can be merged in

**Pull request merged**